tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -503,17 +503,55 @@ where
         Self::new_with_connection_cache(rpc_client, websocket_url, config, connection_cache).await
     }
 
+    /// Try to create LeaderTpuService
+    /// Retries until successful or timeout is reached
+    async fn try_create_leader_tpu_service(
+        rpc_client: Arc<RpcClient>,
+        websocket_url: &str,
+        exit: Arc<AtomicBool>,
+        timeout_seconds: u64,
+    ) -> Result<LeaderTpuService> {
+        let start = tokio::time::Instant::now();
+        loop {
+            if start.elapsed().as_secs() > timeout_seconds {
+                return Err(TpuSenderError::Custom(format!(
+                    "Failed to create LeaderTpuService within {timeout_seconds}s timeout"
+                )));
```

Comment:
Could it be useful for the users to see the `websocket_url` in the error message?

```suggestion
                return Err(TpuSenderError::Custom(format!(
                    "Failed to create LeaderTpuService, connecting to: {websocket_url}, timout: {timeout_seconds}"
                )));
```